src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -482,6 +482,11 @@ public ProcessStartInfo StartInfo
             {
                 if (_startInfo == null)
                 {
+                    if (Associated && Process.GetCurrentProcessId() != Id)
```

Comment:
Actually, looking deeper, this logic looks wrong. __Associated__ returns true after you've _started_ the Process:
'''
Process process = new Process();
process.StartInfo.UseShellExecute = false;
process.StartInfo.FileName = "notepad.exe";
process.Start();

Console.WriteLine(process.StartInfo.FileName); // This now throws

```
You should only throw when you did not start the process yourself.